### PR TITLE
Documentation for mod_auth/forms.py

### PR DESCRIPTION
**Pull request by pranavsid98**
Documentation for mod_auth/forms.py

### CODE REVIEW

File: mod_auth/forms.py [hunk 1]
```diff
@@ -33,12 +38,25 @@ def _email_not_in_use(form, field):
...
     role = Role.query.filter(Role.id == field.data).first()
     if role is None:
         raise ValidationError('Role id is invalid')
 
 
 class LoginForm(Form):
+	'''
+	Form for Login
+
+	attr email: user's email
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: This documentation doesn't provide any useful information (attr) to the user, and for an IDE it's not needed either.

Comment (pranavsid98) [inline mod_auth/forms.py]: Did you mean that this particular Login Form didnt require documentation? Or does the mod_auth/forms.py file not require documentation as a whole?
Thanks

Comment (canihavesomecoffee) [inline mod_auth/forms.py]: A class description can be added, but the attr lines are of no use. A reader of the code can see the attributes right below the comment, and an IDE doesn't need them to show the attributes for auto-completion.

**Comment by pranavsid98** [comment]
Requested changes made

File: mod_auth/forms.py [hunk 2]
```diff
@@ -7,13 +7,18 @@
 
 
 def unique_username(form, field):
-    # Check if a user already exists with this name
+    '''
+		Check if a user already exists with this name
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: It's even worse now.

File: mod_auth/forms.py [hunk 3]
```diff
@@ -7,13 +7,18 @@
 
 
 def unique_username(form, field):
-    # Check if a user already exists with this name
+    '''
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: Double quotes please...

File: mod_auth/forms.py [hunk 4]
```diff
@@ -7,13 +7,18 @@
...
+	'''
     user = User.query.filter(User.name == field.data).first()
     if user is not None:
         raise ValidationError('There is already a user with this name')
 
 
 def valid_password(form, field):
+	'''
+	Function to check for validity of a password
+	'''
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: Where's the parameter explanations?

**Comment by pranavsid98** [comment]
Is it better now? Changes made.

File: mod_auth/forms.py [hunk 5]
```diff
@@ -7,18 +7,24 @@
 
 
 def unique_username(form, field):
-    '''
-		Check if a user already exists with this name
-	'''
+    """
```
Comment (satyammittal) [inline mod_auth/forms.py]: Please Check Indentation.

**Comment by pranavsid98** [comment]
@satyammittal 
I was following that layout initially. But, conforming to PEP 257, I changed it to double quotes. I changed the indentation. Any other suggestions?

File: mod_auth/forms.py [hunk 6]
```diff
@@ -28,7 +34,10 @@ def valid_password(form, field):
 
 
 def email_not_in_use(has_user_field=False):
-    def _email_not_in_use(form, field):
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: Why is this line removed? It's quite essential....

File: mod_auth/forms.py [hunk 7]
```diff
@@ -7,13 +7,24 @@
 
 
 def unique_username(form, field):
-    # Check if a user already exists with this name
+    	"""
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: Alignment is still not correct... Compare to the line with the comment that's gone now...

File: mod_auth/forms.py [hunk 8]
```diff
@@ -7,13 +7,24 @@
...
+    	Check if a user already exists with this name
+
+	:param field: The data value for the 'name' inserted by new User
+	:type field.data : str
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: I'd rather see the type of field than field.data

File: mod_auth/forms.py [hunk 9]
```diff
@@ -23,7 +34,10 @@ def valid_password(form, field):
 
 
 def email_not_in_use(has_user_field=False):
-    def _email_not_in_use(form, field):
+	"""
+	Function to check if the passed email is already in use.
+
```
Comment (canihavesomecoffee) [inline mod_auth/forms.py]: Missing parameter

**Comment by canihavesomecoffee** [comment]
And the layout of the mailer is factually pretty ok, but I missed out on the ' vs ". It definitely must be double quotes.

File: mod_auth/forms.py (hunk 6 above)
Comment (pranavsid98) [inline mod_auth/forms.py]: Oh damn. My bad. Sorry. Didnt notice that I removed it.

File: mod_auth/forms.py (hunk 7 above)
Comment (pranavsid98) [inline mod_auth/forms.py]: Im not really sure why this alignment error is happening. I gave it exactly one tab space

**Comment by pranavsid98** [comment]
Requested changes made

**Comment by canihavesomecoffee** [comment]
It's fine, apart from the alignment. I'll correct that myself :)
